inference-engine/src/mkldnn_plugin/nodes/ctc_greedy_decoder.cpp

```diff
@@ -0,0 +1,157 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "base.hpp"
+#include "ie_parallel.hpp"
+
+#include <vector>
+#include <string>
+
+namespace InferenceEngine {
+namespace Extensions {
+namespace Cpu {
+
+class CTCGreedyDecoderImpl: public ExtLayerBase {
+public:
+    explicit CTCGreedyDecoderImpl(const CNNLayer* layer) : mergeRepeated_(true) {
+        std::string errPrefix = "CTCGreedyDecoder layer with name '" + layer->name + "' ";
+        if (layer->insData.size() != 2)
+            THROW_IE_EXCEPTION << errPrefix << "has invalid number of input edges: " << layer->insData.size();
+        if (layer->outData.size() != 1)
+            THROW_IE_EXCEPTION << errPrefix << "has invalid number of outputs edges: " << layer->outData.size();
+
+        auto inData = layer->insData[DATA_INDEX].lock();
+        auto sequenceLenData = layer->insData[SEQUENCE_LENGTH_INDEX].lock();
+        if (!inData || !sequenceLenData)
+            THROW_IE_EXCEPTION << errPrefix << "has nullable inputs.";
+        if (inData->getTensorDesc().getDims()[0] != sequenceLenData->getTensorDesc().getDims()[0] &&
+                inData->getTensorDesc().getDims()[1] != sequenceLenData->getTensorDesc().getDims()[1])
+            THROW_IE_EXCEPTION << errPrefix << "has invalid input shapes.";
+        if (inData->getTensorDesc().getPrecision() != Precision::FP32 &&
+                inData->getTensorDesc().getPrecision() != Precision::BF16)
+            THROW_IE_EXCEPTION << errPrefix << "has unsupported 'data' input precision: " << inData->getTensorDesc().getPrecision();
+        if (sequenceLenData->getTensorDesc().getPrecision() != Precision::FP32)
+            THROW_IE_EXCEPTION << errPrefix << "has unsupported 'sequence_length' input precision: " << sequenceLenData->getTensorDesc().getPrecision();
+
+        std::vector<DataConfigurator> inputConfigs{{ConfLayout::PLN, Precision::FP32}, {ConfLayout::PLN, Precision::FP32}};
+        std::vector<DataConfigurator> outputConfigs{{ConfLayout::PLN, Precision::FP32}};
+        addConfig(layer, inputConfigs, outputConfigs);
+
+        if (layer->CheckParamPresence("ctc_merge_repeated")) {
+            mergeRepeated_ = layer->GetParamAsBool("ctc_merge_repeated");
+        } else if (layer->CheckParamPresence("merge_repeated")) {
+            mergeRepeated_ = layer->GetParamAsBool("merge_repeated", true);
+        }
+    }
+
+    StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs,
+                       ResponseDesc *resp) noexcept override {
+        const float* probabilities = inputs[DATA_INDEX]->cbuffer().as<const float*>() +
+            inputs[DATA_INDEX]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        const float* sequenceLengths = inputs[SEQUENCE_LENGTH_INDEX]->cbuffer().as<const float*>() +
```

Comment:
As for me `sequenceMask` would be more expressive name here and allow not mix it up with `sequenceLengthB` (which actually can be renamed in just sequenceLength)

Reply:
Fixed